docs/src/main/asciidoc/amqp.adoc

```diff
@@ -44,194 +56,261 @@ The solution is located in the `amqp-quickstart` {quickstarts-tree-url}/amqp-qui
+Now let's consume the quote request and give out a price.
+Inside the `processor` project, locate the `src/main/java/org/acme/amqp/processor/QuoteProcessor.java` file and add the following:
 
 [source, java]
 ----
-package org.acme.amqp;
+package org.acme.amqp.processor;
 
-import io.smallrye.reactive.messaging.annotations.Channel;
-import org.reactivestreams.Publisher;
+import java.util.Random;
 
-import javax.inject.Inject;
-import javax.ws.rs.GET;
-import javax.ws.rs.Path;
-import javax.ws.rs.Produces;
-import javax.ws.rs.core.MediaType;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.acme.amqp.model.Quote;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+
+import io.smallrye.reactive.messaging.annotations.Blocking;
 
 /**
- * A simple resource retrieving the "in-memory" "my-data-stream" and sending the items as server-sent events.
+ * A bean consuming data from the "request" AMQP queue and giving out a random quote.
+ * The result is pushed to the "quotes" AMQP queue.
  */
-@Path("/prices")
-public class PriceResource {
-
-    @Inject
-    @Channel("my-data-stream") Publisher<Double> prices;         // <1>
+@ApplicationScoped
+public class QuoteProcessor {
 
-    @GET
-    @Produces(MediaType.TEXT_PLAIN)
-    public String hello() {
-        return "hello";
-    }
+    private Random random = new Random();
 
-    @GET
-    @Path("/stream")
-    @Produces(MediaType.SERVER_SENT_EVENTS)                     // <2>
-    public Publisher<Double> stream() {                         // <3>
-        return prices;
+    @Incoming("requests")       // <1>
+    @Outgoing("quotes")         // <2>
+    @Blocking                   // <3>
+    public Quote process(String quoteRequest) throws InterruptedException {
+        // simulate some hard working task
+        Thread.sleep(200);
+        return new Quote(quoteRequest, random.nextInt(100));
     }
 }
 ----
-<1> Injects the `my-data-stream` channel using the `@Channel` qualifier
-<2> Indicates that the content is sent using `Server Sent Events`
-<3> Returns the stream (_Reactive Stream_)
+<1> Indicates that the method consumes the items from the `requests` channel
+<2> Indicates that the objects returned by the method are sent to the `quotes` channel
+<3> Indicates that the processing is _blocking_ and cannot be run on the caller thread.
 
-== Configuring the AMQP connector
+The `process` method is called for every AMQP message from the `quote-requests` queue, and will send a `Quote` object to the `quotes` queue.
 
-We need to configure the AMQP connector. This is done in the `application.properties` file.
-The keys are structured as follows:
+As with the previous example we need to configure the connectors in the `application.properties` file.
+Open the `src/main/resources/application.properties` file and add:
+
+[source, properties]
+----
```

Comment:
Add `%dev.quarkus.http.port=8081` to avoid port clash. 
The processor doesn't have an exposed endpoint but in dev mode it is used by the dev ui.
Alternatively we can disable the dev ui.

Reply:
I didn't have the issue.

Reply:
Nevermind. I was running the same app twice..